sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java

```diff
@@ -319,6 +320,13 @@ public final boolean producesSortedKeys(PipelineOptions options) throws Exceptio
     return sourceDelegate.producesSortedKeys(options);
   }
 
+  @Override
+  public void populateDisplayData(DisplayData.Builder builder) {
+    builder
+        .include(sourceDelegate)
+        .add("compressionMode", channelFactory.toString());
```

Comment:
GZIP and BZIP are enum members which implement the class, so their `getClass()` returns something like `com.google.cloud.dataflow.sdk.io.CompressedSource$CompressionMode$1`

I've modified it to check whether the implementing class is an enum, and if so use it's `.name()`.